Mathlib/Analysis/Calculus/ContDiff/RCLike.lean

```diff
@@ -147,6 +147,15 @@ lemma ContDiff.locallyLipschitz {f : E' → F'} (hf : ContDiff 𝕂 1 f) : Local
   rcases hf.contDiffAt.exists_lipschitzOnWith with ⟨K, t, ht, hf⟩
   use K, t
 
+/-- A `C¹` function is Lipschitz on each convex compact set. -/
+theorem ContDiff.lipschitzOnWith {s : Set E} {f : E → F} {n} (hf : ContDiff ℝ n f) (hn : 1 ≤ n)
```

Comment:
I'd rather have a version with `ContDiffOn`, assuming that `f` is smooth only on the convex set `s`, and then deduce the version with `ContDiff`. Also, for coherence with the other lemmas, it should probably be `ContDiff.exists_lipschitzOnWith`.

Reply:
Sounds fair, but this might be harder in practice: I just tried this refactoring, and
- the `ContDiffOn` version seemingly requires `UniqueDiffOn R s`
- specialising this to `ContDiff` means the condition `UniqueDiffOn R s` is also necessary, which it is not, mathematically
(See the most recent pushed commit.)

What do you think, is there a better way?